src/incubator/toast/index.tsx

```diff
@@ -0,0 +1,273 @@
+import {useDidUpdate} from '../../hooks';
+import {Colors, BorderRadiuses, Spacings, Typography, Shadows} from 'style';
+import {Constants} from 'helpers';
+import View from '../../components/view';
+import Text from '../../components/text';
+import Image from '../../components/image';
+import Button from '../../components/button';
+import PanView from '../panView';
+import {ToastProps} from './types';
+import useToastTimer from './helpers/useToastTimer';
+import useToastPresets from './helpers/useToastPresets';
+import useToastAnimation from './helpers/useToastAnimation';
+
+const THRESHOLD = {x: Constants.screenWidth / 4, y: 10};
+
+const Toast = (props: PropsWithChildren<ToastProps>) => {
+  const {
+    visible,
+    position = 'bottom',
+    zIndex,
+    elevation,
+    style,
+    containerStyle,
+    message,
+    messageStyle,
+    renderAttachment,
+    centerMessage,
+    showLoader,
+    action,
+    swipeable,
+    backgroundColor,
+    onDismiss,
+    children,
+    testID
+  } = props;
+
+  const directions = useRef([
+    props.position === 'bottom' ? PanView.directions.DOWN : PanView.directions.UP,
+    PanView.directions.LEFT,
+    PanView.directions.RIGHT
+  ]);
+
+  const viewRef = useRef();
+  const [toastHeight, setToastHeight] = useState(500);
+
+  const {clearTimer, setDismissTimer} = useToastTimer(props);
+  const {icon, iconColor, accessibilityMessage} = useToastPresets(props);
+
+  const playAccessibilityFeatures = () => {
+    if (visible) {
+      if (viewRef.current && action) {
+        // @ts-expect-error
+        const reactTag = findNodeHandle(viewRef.current);
+        AccessibilityInfo.setAccessibilityFocus(reactTag!);
+      } else if (message) {
+        AccessibilityInfo.announceForAccessibility?.(accessibilityMessage);
+      }
+    }
+  };
+
+  const {isAnimating, toggleToast, toastOpacity, toastTranslateY} = useToastAnimation({
+    ...props,
+    toastHeight,
+    setDismissTimer,
+    playAccessibilityFeatures
+  });
+
+  useEffect(() => {
+    if (visible) {
+      toggleToast(visible, {delay: 100});
+    }
+    return () => clearTimer();
+  }, []);
+
+  useDidUpdate(() => {
+    if (!visible) {
+      clearTimer();
+    }
+
+    toggleToast(visible);
+  }, [visible]);
+
+  const handleDismiss = useCallback(() => {
+    clearTimer();
+    onDismiss?.();
+  }, [onDismiss]);
+
+  const positionStyle = useMemo(() => {
+    return {
+      position: 'absolute',
+      left: 0,
+      right: 0,
+      [position]: 0
+    } as Pick<ViewStyle, 'top' | 'left' | 'right' | 'bottom' | 'position'>;
+  }, [position]);
+
+  const onLayout = useCallback(event => {
+    const height = event.nativeEvent.layout.height;
+    if (height !== toastHeight) {
+      setToastHeight(height);
+    }
+  },
+  [toastHeight]);
+
+  const renderRightElement = () => {
+    // NOTE: order does matter
+    if (showLoader) {
+      const loaderColors = [Colors.rgba(Colors.white, 0.3), Colors.rgba(Colors.grey30, 0.7)];
```

Comment:
Is it better to keep this here or move it to a global const?

Reply:
I actually removed it all together, it belong to the private loader anyway